xtask/src/lib.rs

```diff
@@ -548,6 +555,27 @@ pub fn format_package(workspace: &Path, package: Package, check: bool) -> Result
     Ok(())
 }
 
+/// Check for unused dependencies across the entire `esp-hal` workspace using `cargo machete`.
+pub fn check_unused_deps() -> Result<()> {
+    log::info!("Checking unused dependencies across `esp-hal` workspace");
+
+    let status = Command::new("cargo-machete")
```

Comment:
I'd much prefer integrating with a libary, then shelling out calls to random binaries (that we aren't installing in CI btw, so I think the CI job will fail). Did you consider using udeps instead? It has a library that we can use in the xtask: https://github.com/est31/cargo-udeps/blob/master/src/lib.rs.

Reply:
With `cargo udeps` all deps seem to have been used 🤔

Reply:
> Maybe we can try to run Check unused deps step only when a Cargo.toml was modified?

Probably not what we want - usually we don't willingly add unused deps but remove the code using them and forget to remove them from Cargo.toml